Jenkinsfile_k8s

```diff
@@ -1,18 +1,178 @@
-#!/usr/bin/env groovy
+pipeline {
+  agent {
+    kubernetes {
+      inheritFrom 'jnlp-linux'
+      yaml '''
+apiVersion: "v1"
+kind: "Pod"
+metadata:
+  labels:
+    jenkins: "agent"
+    job: "jenkins-io"
+spec:
+  tolerations:
+    - key: "os"
+      operator: "Equal"
+      value: "linux"
+      effect: "NoSchedule"
+  affinity:
+    nodeAffinity:
+      requiredDuringSchedulingIgnoredDuringExecution:
+        nodeSelectorTerms:
+          - matchExpressions:
+              - key: kubernetes.io/os
+                operator: In
+                values:
+                  - linux
+  restartPolicy: "Never"
+  containers:
+    - name: "node"
+      image: "node:16.13.1"
+      securityContext:
+        privileged: false
+      tty: true
+      command:
+      - cat
+    - name: "ruby"
+      image: "ruby:2.6.9"
+      securityContext:
+        privileged: false
+      tty: true
+      command:
+      - cat
+    - name: "blobxfer"
+      image: "mcr.microsoft.com/blobxfer:1.9.1"
+      command:
+      - cat
+      securityContext:
+        privileged: false
+      tty: true
+      '''
+    }
+  }
+
+  environment {
+    TZ = "UTC"
+    USE_LOCAL_NODE = "true"
+    USE_LOCAL_RUBY = "true"
+    HOME = "/tmp"
+  }
+
+  triggers {
+    cron("${env.BRANCH_NAME == 'master' ? 'H/30 * * * *' : ''}")
+  }
+
+  options {
+    timeout(time: 60, unit: 'MINUTES')
+    ansiColor('xterm')
+    disableConcurrentBuilds()
+    buildDiscarder logRotator(artifactDaysToKeepStr: '', artifactNumToKeepStr: '', daysToKeepStr: '5', numToKeepStr: '5')
+  }
 
-/* Only keep the 10 most recent builds. */
-def projectProperties = [
-    [$class: 'BuildDiscarderProperty',strategy: [$class: 'LogRotator', numToKeepStr: '5']],
-]
+  stages {
+    stage('NPM Install') {
+      steps {
+        container('node') {
+          sh 'npm install'
+        }
+      }
+    }
 
-def principalBranch = 'master'
+    stage('Bundle Install') {
+      steps {
+        container('ruby') {
+          // throw errors if Gemfile has been modified since Gemfile.lock
+          sh 'bundle config --global frozen 1'
+          sh 'bundle install'
+        }
+      }
+    }
 
-/* Start by building the image, and eventually pushing it if the execution is on the master branch */
-buildDockerAndPublishImage('jenkinsio', [mainBranch: principalBranch])
+    stage('Build') {
+      steps {
+        container('node') {
+          sh 'make assets'
+        }
+        container('ruby') {
+          sh 'make fetch'
+        }
+        container('ruby') {
+          sh 'make real_generate'
+        }
+      }
+    }
 
-if (env.BRANCH_NAME == principalBranch) {
-  withCredentials([string(credentialsId: 'updatecli-github-token', variable: 'UPDATECLI_GITHUB_TOKEN')]) {
+    /*
```

Comment:
?

Reply:
Right now a different ci is deploying to azure, and the secrets arnt in infra.

Reply:
o so just remove then?

Reply:
Probably, but I also want to chat with people after holidays. I've marked the pr as do not merge for now